pimd/pim_sock.c

```diff
@@ -220,6 +191,93 @@ int pim_socket_mcast(int protocol, struct in_addr ifaddr, struct interface *ifp,
 
 	if (setsockopt(fd, IPPROTO_IP, IP_MULTICAST_IF, (void *)&mreq,
 		       sizeof(mreq))) {
+	    zlog_warn(
+		    "Could not set Outgoing Interface Option on socket fd=%d: errno=%d: %s",
+		    fd, errno, safe_strerror(errno));
+	    close(fd);
+	    return PIM_SOCK_ERR_IFACE;
+	}
+
+	if (setsockopt(fd, SOL_SOCKET, SO_RCVBUF, &rcvbuf, sizeof(rcvbuf)))
```

Comment:
setsockopt for option : SO_RCVBUF looks same for IPV6 as well. Shall we keep this inside pim_socket_mcast, to keep it common for V4 & V6?